mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/GroupConsumerRetry.java

```diff
@@ -140,6 +93,28 @@ String asString() {
         LOGGER.error("Exception happens when handling event {} of user {}",
             event.getEventId().getId().toString(), event.getUser().asString(), throwable);
 
-        return retryPublisher.publish(event, eventAsBytes, currentRetryCount);
+        return retryOrStoreToDeadLetter(event, eventAsBytes, currentRetryCount);
+    }
+
+    private Mono<Void> retryOrStoreToDeadLetter(Event event, byte[] eventAsByte, int currentRetryCount) {
+        if (currentRetryCount >= retryBackoff.getMaxRetries()) {
+            return eventDeadLetters.store(group, event);
+        }
+        return sendRetryMessage(event, eventAsByte, currentRetryCount);
+    }
+
+    private Mono<Void> sendRetryMessage(Event event, byte[] eventAsByte, int currentRetryCount) {
+        Mono<OutboundMessage> retryMessage = Mono.just(new OutboundMessage(
+            retryExchangeName.asString(),
+            EMPTY_ROUTING_KEY,
+            new AMQP.BasicProperties.Builder()
+                .headers(ImmutableMap.of(RETRY_COUNT, currentRetryCount + 1))
+                .build(),
+            eventAsByte));
+
+        return sender.send(retryMessage)
+            .doOnError(throwable -> LOGGER.error("Exception happens when publishing event of user {} to retry exchange," +
```

Comment:
could you add name of retry exchange and may be event id in the log?

Reply:
-> For the eventID there is a full task coming about context logging.
-> I think the group makes more sens than the `retry exchange`. BTW it do also belong to context logging IMO

Also note that I did not modify the logging line that you did write, just I moved it around.